Remove only works with the last peer of the list

[This code](https://github.com/mailgun/groupcache/blob/master/groupcache.go#L268) captures only the last iteration of the for loop. When the goroutines finally run the variable `peer` has already changed to the last value.

```go
		// Asynchronously clear the key from all hot and main caches of peers
		for _, peer := range g.peers.GetAll() {
			// avoid deleting from owner a second time
			if peer == owner {
				continue
			}

			wg.Add(1)
			go func() {
				errs <- g.removeFromPeer(ctx, peer, key)
				wg.Done()
			}()
		}
```

You can see the same efect here: https://play.golang.org/p/I82KNfRkqSP 
Go vet will also highlight the error:
```
./prog.go:16:19: loop variable peer captured by func literal
```

The solution here is easy though. The smallest working change is to pass the peer as an argument to the goroutine: https://play.golang.org/p/0mNV3nSkGUy

--- a/groupcache.go
+++ b/groupcache.go
@@ -265,7 +265,7 @@ func (g *Group) Remove(ctx context.Context, key string) error {
 			}
 
 			wg.Add(1)
-			go func(peer string) {
+			go func(peer ProtoGetter) {
 				errs <- g.removeFromPeer(ctx, peer, key)
 				wg.Done()
 			}(peer)

--- a/groupcache.go
+++ b/groupcache.go
@@ -265,10 +265,10 @@ func (g *Group) Remove(ctx context.Context, key string) error {
 			}
 
 			wg.Add(1)
-			go func() {
+			go func(peer string) {
 				errs <- g.removeFromPeer(ctx, peer, key)
 				wg.Done()
-			}()
+			}(peer)
 		}
 		go func() {
 			wg.Wait()

